drivers/SmartThings/matter-button/src/init.lua

```diff
@@ -3,172 +3,222 @@ local log = require "log"
-      device:emit_event(capabilities.button.supportedButtonValues({"pushed", "held"},{visibility = {displayed = false}})) --supported values
+    local MSR = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.SwitchFeature.MOMENTARY_SWITCH_RELEASE})
+    local MSL = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.SwitchFeature.MOMENTARY_SWITCH_LONG_PRESS})
+    local MSM = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.SwitchFeature.MOMENTARY_SWITCH_MULTI_PRESS})
 
-    elseif #MSM > 0 then
-      device:set_field(BUTTON_TYPE, BUTTON_TYPE_MAP.MULTI_PRESS) -- multipress
+    -- We have a static profile that will work for this number of buttons
+    if contains(STATIC_PROFILE_SUPPORTED, #MS) then
+      device:try_update_metadata({profile = string.format("%d-button", #MS)})
+    end
 
-      --send read MultiPressMax attribute request
-      local req = clusters.Switch.attributes.MultiPressMax:read(device)
-      device:send(req)
+    -- At the moment, we're taking it for granted that all momentary switches only have 2 positions
+    -- TODO: flesh this out for NumberOfPositions > 2
+    local current_component_number = 2
+    for _, ep in ipairs(MS) do -- for each momentary switch endpoint (including main)
+      -- build the mapping of endpoints to components if we have a static profile (multi-component)
+      if contains(STATIC_PROFILE_SUPPORTED, #MS) then
+        local map = device:get_field(ENDPOINT_TO_COMPONENT_MAP) or {}
+        if ep ~= main_endpoint then
+          map[ep] = string.format("button%d", current_component_number)
+          current_component_number = current_component_number + 1
+        else
+          map[ep] = "main"
+        end
+        device:set_field(ENDPOINT_TO_COMPONENT_MAP, map, {persist = true})
+      else -- use parent/child
+        if ep ~= main_endpoint then -- don't create a child device that maps to the main endpoint
+          local name = string.format("%s %d", device.label, current_component_number)
+          driver:try_create_device(
+            {
+              type = "EDGE_CHILD",
+              label = name,
+              profile = "child-button",
+              parent_device_id = device.id,
+              parent_assigned_child_key = string.format("%02X", ep),
+              vendor_provided_label = name
+            }
+          )
+          current_component_number = current_component_number + 1
+        end
+      end
 
+      -- this ordering is important, as MSL & MSM devices must also support MSR
+      if contains(MSL, ep) then
+        device:emit_event_for_endpoint(ep, capabilities.button.supportedButtonValues({"pushed", "held"}, {visibility = {displayed = false}}))
+      elseif contains(MSM, ep) then
+        -- ask the device to tell us its max number of presses
+        device:send(clusters.Switch.attributes.MultiPressMax:read(device, ep))
```

Comment:
I'm going to leave this in for multipressmax for two reasons:

1. worried about race conditions because the endpoint_to_component map might not be set up
2. in the case we're using child devices, it doesn't seem like each endpoint that supports the attribute would be queried

So I'll just remove it from the subscription table to compensate. How does that sound?